js_modules/dagster-ui/packages/ui-core/src/assets/LineageDepthControl.tsx

```diff
@@ -0,0 +1,67 @@
+import {Box, Button, Colors, Icon, JoinedButtons, TextInput} from '@dagster-io/ui-components';
+import React, {useEffect, useState} from 'react';
+
+export const LineageDepthControl = ({
+  value,
+  max,
+  onChange,
+}: {
+  value: number;
+  max: number;
+  onChange: (v: number) => void;
+}) => {
+  const [text, setText] = useState(`${value}`);
+  useEffect(() => {
```

Comment:
useLayoutEffect? It would at least skip a full re-render loop 😬

Reply:
Oh interesting! I didn't think about that - will change this